src/routes/product-detail/index.tsx

```diff
@@ -1,26 +1,43 @@
-import { Grid, Container, Box, Text, Flex } from "@mantine/core"
+import { useAtom } from "jotai"
+import { useEffect } from "react"
 import { useQuery } from "@tanstack/react-query"
+import { navigate } from "wouter/use-browser-location"
+import { Grid, Container, Box, Text, Flex } from "@mantine/core"
 
 import { ProductDetailCard } from "./ProductDetailCard"
 import { ProductDetailInsights } from "./ProductDetailInsights"
 
+import { siteAtom } from "../../store/site"
 import { truncate } from "../../utils/truncate"
-
 import { getProductById } from "../../utils/query-product"
 import { FullPageLoader } from "../../components/Loader"
 
+import { SITES } from "../../constants/sites"
+
 interface Props {
   id: string
 }
 
 export const ProductDetailPage = ({ id }: Props) => {
+  const [siteKey] = useAtom(siteAtom)
+
   const query = useQuery({
     queryKey: ["product", id ?? null],
     queryFn: ({ queryKey: [_, id] }) => getProductById(parseInt(id, 10)),
   })
 
   const product = query.data
 
+  // If the product is from a different shop, redirect back to the product listing page.
+  // This prevents the user from seeing data from a different shop.
+  useEffect(() => {
+    const site = SITES.find((s) => s.key === siteKey)
+
+    if (product && site && site.shopId !== product.shopId) {
+      navigate("/admin/products")
+    }
+  }, [siteKey, product])
```

Comment:
This is a bug in the existing Shoppy app. Go to current demo at https://metabase-shoppy.vercel.app, try to open a product and try to switch site. You'll see that we can still see products that are supposed to be in another tenant.

This makes it so if they open a URL with the product from the wrong tenant, they'll get redirected to the product listing page.